lib/game_teams_methods.rb

```diff
@@ -27,24 +30,22 @@ def worst_offense_team
   end
 
   def assign_goals_by_teams
-    goals = @game_teams_table['goals']
-    team_goals = Hash.new
-    @game_teams_table['team_id'].each.with_index do |id, idx|
-      if team_goals.has_key?(id)
-        team_goals[id] << goals[idx]
-      else
-        team_goals[id] = [goals[idx]]
-      end
+    team_goals = Hash.new { |hash, key| hash[key] = []}
```

Comment:
I don't quite understand what the block is doing here, isn't it just making a hash?